### PR TITLE
Extract the pattern overrides toolbar indicator from the `block-editor` package

### PR DESCRIPTION
**Pull request by kevin940726**
## What?

Part of https://github.com/WordPress/gutenberg/issues/60619. Address the issue discussed in https://github.com/WordPress/gutenberg/pull/62317#discussion_r1631024218.

Extract out pattern overrides related code from the `block-editor` package to the `patterns` package, specifically the toolbar indicator.

## Why?

Pattern Overrides is a WP-only feature so ideally it doesn't belong in the `block-editor` package.

## How?

Create a `PatternOverridesBlockControls` component and conditionally render it using the pattern-overrides hook.

## Testing Instructions

1. Create a pattern with overrides and insert it into a post.
2. Select an overridable block and expect the toolbar indicator to show (see below image).
3. Select multiple overridable blocks and expect the toolbar indicator to show (see below image).
4. Selecting blocks inside the pattern focus mode should show a different indicator (see below image).

## Screenshots or screencast 

Select a single pattern overrides block:
image

Select multiple pattern overrides blocks:
image

Select a block inside the pattern focus mode:
image

### CODE REVIEW

File: packages/block-editor/src/components/pattern-overrides-toolbar-indicator/index.js
Comment (kevin940726) [inline packages/block-editor/src/components/pattern-overrides-toolbar-indicator/index.js:1]: Moved to `packages/patterns/src/components/pattern-overrides-block-controls.js`.

File: packages/block-editor/src/components/pattern-overrides-toolbar-indicator/style.scss
Comment (kevin940726) [inline packages/block-editor/src/components/pattern-overrides-toolbar-indicator/style.scss:1]: Moved to `packages/patterns/src/components/style.scss`.

File: packages/editor/src/hooks/pattern-overrides.js [hunk 1]
```diff
@@ -43,6 +44,7 @@ const withPatternOverrideControls = createHigherOrderComponent(
 				{ props.isSelected && isSupportedBlock && (
 					<ControlsWithStoreSubscription { ...props } />
 				) }
+				{ isSupportedBlock && <PatternOverridesBlockControls /> }
```
Comment (talldan) [inline packages/editor/src/hooks/pattern-overrides.js:47]: It's a shame there's no way to add a condition like `( props.isSelected || props.isMultiselected )` here

**Review by talldan — APPROVED** [review]
Code looks good and it tests well, thanks for addressing this.

**Comment by ellatrix** [comment]
I just cherry-picked this PR to the wp/6.6-beta-3 branch to get it included in the next release: 8d8f6818b4

File: packages/patterns/src/components/pattern-overrides-block-controls.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+			);
+			const _hasParentPattern = selectedClientIds.every(
+				( clientId ) =>
+					getBlockParentsByBlockName( clientId, 'core/block', true )
+						.length > 0
+			);
+			return {
+				clientIds: selectedClientIds,
+				hasPatternOverrides: _hasPatternOverrides,
+				hasParentPattern: _hasParentPattern,
```
Comment (Mamaduka) [inline packages/patterns/src/components/pattern-overrides-block-controls.js:144]: I am unsure if this affects anything substantial, but these values will be `true` when `selectedClientIds` is an empty array. Do to how `Array.every` works in JS.

File: packages/editor/src/hooks/pattern-overrides.js (hunk 1 above)
Comment (Mamaduka) [reply packages/editor/src/hooks/pattern-overrides.js:47]: We should probably add a comment there to avoid accidental regression.

**Pull request merged**